card_data.py

```diff
@@ -440,29 +405,41 @@ def removeUnreleasedCards(cards):
     "Witcher": "Witcher"
 }
 
-LOCALES = ["en-US", "de-DE", "es-ES", "es-MX", "fr-FR", "it-IT", "ja-JP", "pl-PL", "pt-BR", "ru-RU", "zh-CN", "zh-TW"]
-
-TEMPLATES = getCardTemplates()
-ABILITIES = getAbilityData()
+IMAGE_URL = ""
+IMAGE_SIZES = ['original', 'high', 'medium', 'low', 'thumbnail']
+TEMPLATES = {}
+ABILITIES = {}
 TOOLTIPS = {}
-TOOLTIP_DATA = getTooltipData()
+TOOLTIP_DATA = {}
 CARD_NAMES = {}
 FLAVOR_STRINGS = {}
 
-for region in LOCALES:
-    TOOLTIPS[region] = getRawTooltips(region)
-    CARD_NAMES[region] = getCardNames(region)
-    FLAVOR_STRINGS[region] = getFlavorStrings(region)
-
-cardData = createCardJson()
-
-# Requires information about other cards, so needs to be done after we have looked at every card.
-evaluateInfoData(cardData)
-# We have to do this as well to catch cards like Botchling, that are explicitly named in the Baron's tooltip.
-evaluateTokens(cardData)
-# After the info text has been evaluated, we can extract the keywords (e.g. deploy).
-evaluateKeywords(cardData)
-removeInvalidImages(cardData)
-removeUnreleasedCards(cardData)
-# Save under v0-9-10_2017-09-05.json if the script is ran on 5 September 2017 with patch v0-9-10.
-saveJson(PATCH + "_" + datetime.utcnow().strftime("%Y-%m-%d") + ".json", cardData)
+def createCardJson(rawFolder, patch):
+    # Replace with these values {0} : card id, {1} : variation id, {0} : image size
+    global IMAGE_URL
+    IMAGE_URL = "https://firebasestorage.googleapis.com/v0/b/gwent-9e62a.appspot.com/o/images%2F" + patch + "%2F{0}%2F{1}%2F{2}.png?alt=media"
+
+    global TEMPLATES
+    TEMPLATES = getCardTemplates(rawFolder)
+    global ABILITIES
+    ABILITIES = getAbilityData(rawFolder)
+    global TOOLTIP_DATA
+    TOOLTIP_DATA = getTooltipData(rawFolder)
+
+    for region in utils.LOCALES:
+        TOOLTIPS[region] = getRawTooltips(rawFolder, region)
+        CARD_NAMES[region] = getCardNames(rawFolder, region)
+        FLAVOR_STRINGS[region] = getFlavorStrings(rawFolder, region)
+
+    cardData = createBaseCardJson()
+
+    # Requires information about other cards, so needs to be done after we have looked at every card.
+    evaluateInfoData(cardData)
```

Comment:
Here is your problem. These methods all rely on side-effects, when they should instead be pure-functions.